lib/components/CdeModal.tsx

```diff
@@ -19,23 +29,29 @@ const CdeModal: FC = () => {
         }
     }, [errorMessage, setErrorMessage]);
 
-    const renderStepComponent = () => {
+    const renderStepComponent = (): ReactElement => {
         switch (step) {
             case STEPS.HOME:
                 return <Home/>;
-            case STEPS.REPOSITORY:
+            case STEPS.COLLECTION:
                 return <MappingStep/>;
-            // Add cases for other steps
+            // Add cases for other steps with commonProps
             default:
                 return <div>Unknown step</div>;
         }
     };
 
-
     return (
         <>
-            <Modal open={isOpen} onClose={handleClose} maxWidth="xl">
-                {renderStepComponent()}
+            <Modal open={isModalOpen} onClose={onClose} maxWidth="xl" isInfoOpen={isInfoOpen}>
```

Comment:
Moved the loading logic from the ModalLayout to the current component so that it's addressed independently of the Step